### PR TITLE
SU-488 : Archivo de Cartera - Departamento

### PR DESCRIPTION
**Pull request by tabrez-fiter**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Review by fiter-julius-oketayot — APPROVED** [review]

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/codes/domain/CodeValueRepository.java [hunk 1]
```diff
@@ -18,12 +18,39 @@
...
+                    WHEN LEFT(LPAD(code_score::TEXT, 5, '0'), 2)::INTEGER < 10 THEN
+                        LEFT(LPAD(code_score::TEXT, 5, '0'), 2)::INTEGER::VARCHAR
+                    ELSE
+                        LEFT(LPAD(code_score::TEXT, 5, '0'), 2)
+                END AS ciudadCoderScore
+                FROM m_code_value mcv2
+                WHERE code_id = 41
+            ) mv2
+            ON mv2.ciudadCoderScore = mcv.code_score
+            WHERE mcv.code_id = 40 AND mv2.code_score = :codeScore
```
Comment (hagafrank) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/codes/domain/CodeValueRepository.java]: How reliable is this? Using a literal value for the `code_id`. Maybe it's better to use the code name instead. Something like this:
`WHERE mcv.code_id = (SELECT id FROM m_code WHERE name = 'XXX')

**Review by hagafrank — CHANGES_REQUESTED** [review]
I'm concerned about the use of hard-coded id's in queries baked in the Java code. See comment below

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/codes/domain/CodeValueRepository.java (hunk 1 above)
Comment (tabrez-fiter) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/codes/domain/CodeValueRepository.java]: i already update

**Review by hagafrank — APPROVED** [review]
LGTM

**Pull request merged**